test/system/ard/offshore/test_mooring_packing.py

```diff
@@ -0,0 +1,187 @@
+        )
+
+        # build out the problem based on this model
+        self.prob = om.Problem(model)
+        self.prob.setup()
+
+    def test_packing(self):
+
+        # add design variable
+        self.prob.model.add_design_var("spacing_primary", lower=2.0, upper=15.0)
+        self.prob.model.add_design_var("spacing_secondary", lower=2.0, upper=15.0)
+        self.prob.model.add_design_var("angle_orientation", lower=-90.0, upper=90.0)
+        self.prob.model.add_design_var("angle_skew", lower=-90.0, upper=90.0)
+        self.prob.model.add_design_var("phi_platform", lower=-30.0, upper=30.0)
+        self.prob.model.add_constraint(
+            "mooring_constraint.mooring_spacing",
+            units="m",
+            lower=50.0,
+        )
+        self.prob.model.add_constraint(
+            "spacing_constraint.turbine_spacing",
+            units="m",
+            lower=3 * self.modeling_options["turbine"]["geometry"]["diameter_rotor"],
+        )
+        self.prob.model.add_objective("optiwindnet_coll.total_length_cables")
+
+        # configure the driver
+        self.prob.driver = om.ScipyOptimizeDriver(optimizer="SLSQP")
+        self.prob.driver.options["maxiter"] = 50  # short run
+
+        # setup the problem
+        self.prob.setup()
+
+        # set up the variables
+        self.prob.set_val("spacing_primary", 7.0)
+        self.prob.set_val("spacing_secondary", 7.0)
+        self.prob.set_val("angle_orientation", 0.0)
+        self.prob.set_val("angle_skew", 0.0)
+        self.prob.set_val("phi_platform", 0.0)
+        self.prob.set_val("optiwindnet_coll.x_substations", [500.0])
+        self.prob.set_val("optiwindnet_coll.y_substations", [500.0])
+
+        # run the model
+        self.prob.run_model()
+
+        # compute the derivatives
+        totals = self.prob.compute_totals(
+            of=[
+                "mooring_constraint.mooring_spacing",
+                "spacing_constraint.turbine_spacing",
+            ],
+            wrt=[
+                "spacing_primary",
+                "spacing_secondary",
+                "angle_orientation",
+                "angle_skew",
+                "phi_platform",
+            ],
+        )
+
+        # check total derivatives using OpenMDAO's check_totals
+        deriv_check = self.prob.check_totals(
+            of=[
+                "mooring_constraint.mooring_spacing",
+                "spacing_constraint.turbine_spacing",
+            ],
+            wrt=[
+                "spacing_primary",
+                "spacing_secondary",
+                "angle_orientation",
+                "angle_skew",
+                "phi_platform",
+            ],
+            step=1e-6,
+            form="central",
+            show_only_incorrect=False,
+            compact_print=True,
+        )
+
+        # optionally, assert that the analytic and finite difference derivatives are close
+        for out_var, in_var in deriv_check.keys():
```

Comment:
Each iteration of the loop should be a separate subtest (can be done as a loop still) with the test name using the out and in vars to help with debugging.